tests/main/interfaces-calendar-service/task.yaml

```diff
@@ -21,7 +22,19 @@ prepare: |
 restore: |
     kill "$(cat dbus-launch.pid)"
     rm -f dbus-launch.pid
-    rm -rf "$XDG_CONFIG_HOME" "$XDG_DATA_HOME" "$XDG_CACHE_HOME"
+
+    # Retry until the xdg directory can be removed sucessfully
+    # The retry is needed because sporadically it is still locked and
```

Comment:
Locking a directory doesn't block removal. What was the error message you saw?

Reply:
```
travis_time:start:6d57b98c
2018-06-13 06:07:02 Error restoring google:opensuse-42.3-64:tests/main/interfaces-calendar-service : 
-----
++ cat dbus-launch.pid
+ kill 28376
+ rm -f dbus-launch.pid
+ rm -rf /home/gopath/src/github.com/snapcore/snapd/tests/main/interfaces-calendar-service/config /home/gopath/src/github.com/snapcore/snapd/tests/main/interfaces-calendar-service/share /home/gopath/src/github.com/snapcore/snapd/tests/main/interfaces-calendar-service/cache
rm: cannot remove '/home/gopath/src/github.com/snapcore/snapd/tests/main/interfaces-calendar-service/share/gvfs-metadata': Directory not empty
-----
travis_time:end:6d57b98c:start=1528870022608828812,finish=1528870022685011182,duration=76182370
```
Which is quite funny because we're doing `rm -rf`

Reply:
One thing to note, it's `gvfs-metadata`, maybe it's going through fuse?

Reply:
@zyga what it is locked is a file inside the directory, not the directory itself. I am updating the comment